rest/handler.go

```diff
@@ -304,6 +304,30 @@ func (h *handler) checkAuth(context *db.DatabaseContext) (err error) {
 
 	}(time.Now())
 
+	// Is there a TLS client cert?
+	var clientCertEmailAddress *string
+	if clientCerts := h.rq.TLS.VerifiedChains; len(clientCerts) > 0 {
+		clientCert := *clientCerts[0][0]
+		if len(clientCert.EmailAddresses) > 0 {
+			clientCertEmailAddress = &clientCert.EmailAddresses[0]
+		} else if strings.Contains(clientCert.Subject.CommonName, "@") {
+			// WORKAROUND: Use the Common Name if there is no email address present,
+			// until LiteCore can be fixed to create certs with the email address stored correctly
+			clientCertEmailAddress = &clientCert.Subject.CommonName
+		} else {
+			// TODO: Detect this earlier when the connection is made, i.e. in a TLS cert validation callback.
+			return base.HTTPErrorf(http.StatusUnauthorized,
+				"X.509 client cert has no email address to identify the user")
+		}
+		base.Infof(base.KeyHTTP, "Client TLS cert: %s; email = %s",
+			clientCert.Subject.String(), *clientCertEmailAddress)
```

Comment:
Apply log redaction to user email

```suggestion
			clientCert.Subject.String(), base.UD(*clientCertEmailAddress))
```